Wrappers/Python/cil/optimisation/functions/StochasticGradientAlgorithmFunction.py

```diff
@@ -0,0 +1,120 @@
+from cil.optimisation.functions import Function, SumFunction
+import numpy as np
+
+class StochasticGradientAlgorithmFunction(SumFunction):
+    
+    '''Class for use as objective function in Stochastic Gradient type algorithms such as GD
```

Comment:
not `GD` as that is just gradient descent

Reply:
Isn't `GD` the algorithm the one we actually used?

Reply:
what about this
```
Class for use as objective function in stochastic subset algorithms such as Stochastic Gradient Descent

`gradient` returns `subset_gradient` for a randomly chosen subset
```

Reply:
Maybe

```
Class for use as objective function in gradient type algorithms to enable the use of subsets, such as Stochastic Gradient Descent

`gradient` returns `subset_gradient` for a (randomly chosen) subset

Reply:
great, Note that I'm not sure if my second line is appropriate for this class (it is maybe for a derived class)

Reply:
what about
```
Class for use as objective function in gradient type algorithms to enable the use of subsets.

The `gradient` method is implemented in children classes and allows to return an approximation of the gradient based on subset gradients.
```

I removed the reference to Stochastic Gradient Descent as it's not clear to me what this actually means: SGD as a theoretical algorithm, or as an algorithm implemented in CIL?